x/auth/keeper/grpc_query_test.go

```diff
@@ -192,62 +193,120 @@ func (suite *KeeperTestSuite) TestGRPCQueryParameters() {
+	})
+}
+
+func (suite *KeeperTestSuite) TestAddressBytesToString() {
+	var req *types.AddressBytesToStringRequest
+	_, _, addr := testdata.KeyTestPubAddr()
 
 	testCases := []struct {
 		msg       string
 		malleate  func()
 		expPass   bool
-		posttests func(res *types.QueryModuleAccountsResponse)
+		posttests func(res *types.AddressBytesToStringResponse)
 	}{
 		{
 			"success",
 			func() {
-				req = &types.QueryModuleAccountsRequest{}
+				addrBytes := []byte(addr)
+				req = &types.AddressBytesToStringRequest{AddressBytes: addrBytes}
 			},
 			true,
-			func(res *types.QueryModuleAccountsResponse) {
-				var mintModuleExists = false
-				for _, acc := range res.Accounts {
-					var account types.AccountI
-					err := suite.app.InterfaceRegistry().UnpackAny(acc, &account)
-					suite.Require().NoError(err)
-
-					moduleAccount, ok := account.(types.ModuleAccountI)
-
-					suite.Require().True(ok)
-					if moduleAccount.GetName() == "mint" {
-						mintModuleExists = true
-					}
-				}
-				suite.Require().True(mintModuleExists)
+			func(res *types.AddressBytesToStringResponse) {
+				text, err := suite.app.AccountKeeper.ConvertAddressBytesToString(req.AddressBytes)
+				suite.Require().NoError(err)
+				suite.Require().NotNil(text)
+				suite.Require().Equal(text, res.AddressString)
 			},
 		},
 		{
-			"invalid module name",
+			"request is empty",
 			func() {
-				req = &types.QueryModuleAccountsRequest{}
+				req = &types.AddressBytesToStringRequest{}
 			},
-			true,
-			func(res *types.QueryModuleAccountsResponse) {
-				var mintModuleExists = false
-				for _, acc := range res.Accounts {
-					var account types.AccountI
-					err := suite.app.InterfaceRegistry().UnpackAny(acc, &account)
-					suite.Require().NoError(err)
+			false,
+			func(res *types.AddressBytesToStringResponse) {},
+		},
+		{
+			"empty account address in request",
+			func() {
+				emptyAddrBytes := []byte{}
+				req = &types.AddressBytesToStringRequest{AddressBytes: emptyAddrBytes}
+			},
+			false,
+			func(res *types.AddressBytesToStringResponse) {},
+		},
+	}
 
-					moduleAccount, ok := account.(types.ModuleAccountI)
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
+			suite.SetupTest() // reset
 
-					suite.Require().True(ok)
-					if moduleAccount.GetName() == "falseCase" {
-						mintModuleExists = true
-					}
-				}
-				suite.Require().False(mintModuleExists)
+			tc.malleate()
+			ctx := sdk.WrapSDKContext(suite.ctx)
+
+			res, err := suite.queryClient.AddressBytesToString(ctx, req)
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+				suite.Require().NotNil(res)
+			} else {
+				suite.Require().Error(err)
+				suite.Require().Nil(res)
+			}
+
+			tc.posttests(res)
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestAddressStringToBytes() {
+	var req *types.AddressStringToBytesRequest
+	_, _, addr := testdata.KeyTestPubAddr()
+
+	testCases := []struct {
+		msg       string
+		malleate  func()
```

Comment:
I don't think a malleate is useful in this case, you can replace this field with `req`

Reply:
Sure. Done!